test/e2e/framework/volume_util.go

```diff
@@ -174,21 +179,31 @@ func NewGlusterfsServer(cs clientset.Interface, namespace string) (config Volume
 	return config, pod, ip
 }
 
-// iSCSI-specific wrapper for CreateStorageServer.
-func NewISCSIServer(cs clientset.Interface, namespace string) (config VolumeTestConfig, pod *v1.Pod, ip string) {
+// iSCSI-specific wrapper for CreateStorageServer. Note that there can be only
+// one SCSI server running on a node, [Serial] is highly advised.
+func NewISCSIServer(cs clientset.Interface, namespace string) (config VolumeTestConfig, pod *v1.Pod, ip, iqn string) {
+	// Generate cluster-wide unique IQN
+	iqn = fmt.Sprintf(iSCSIIQNTemplate, namespace)
+
 	config = VolumeTestConfig{
 		Namespace:   namespace,
 		Prefix:      "iscsi",
 		ServerImage: imageutils.GetE2EImage(imageutils.VolumeISCSIServer),
-		ServerPorts: []int{3260},
+		ServerArgs:  []string{iqn},
 		ServerVolumes: map[string]string{
 			// iSCSI container needs to insert modules from the host
 			"/lib/modules": "/lib/modules",
+			// iSCSI container needs to configure kernel
+			"/sys/kernel": "/sys/kernel",
```

Comment:
what happens if this is not mounted? (just curious why we did not need it before)

Reply:
iscsiadm was not happy and threw random errors.